### PR TITLE
[PPML] Alibaba ECS Occlum CN Develop guide

### PR DESCRIPTION
**Pull request by qiyuangong**
### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [ ] Application test
- [x] Document test
- [ ] ...

### 5. New dependencies

- [ ] New Python dependencies
       - Dependency1 
       - Dependency2
       - ...
- [ ] New Java/Scala dependencies and their license
       - Dependency1 and license1
       - Dependency2 and license2
       - ...

### CODE REVIEW

File: docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+ 
+图 SGX enclave结构
+作为硬件级的可信执行环境，SGX的攻击面非常小，攻击者即使攻破操作系统和BIOS也无法获取SGX中的应用和数据。但在端到端的应用中，用户还需要确保其他阶段的安全性。简而言之，用户需要确保数据或者文件在SGX外部是加密的，仅在SGX内部被解密和计算，如下图所示。为了实现这个目的，我们往往需要借助密钥管理服务 (Key Management Service, KMS) 的帮助。用户可以将密钥托管到KMS，等应用在SGX中启动后，再从KMS申请和下载密钥。
+ 
+图 SGX应用设计原则
+PPML项目的核心功能是帮助用户迁移现有的应用，用户可以选择迁移现有的大数据AI应用，也可以选择开发全新的应用。PPML应用的开发和常规应用基本相同。例如PySpark的应用代码和常规应用并没有区别。但在设计、编译和部署时有一定的差异。具体表现为：
+* 设计时需要考虑加解密流程，确保明文数据只出现在SGX内部
+* 编译时，需要通过Occlum将应用编译成SGX enclave
+* 部署时，需要将SGX enclave部署到有SGX环境的节点
+在剩下的章节中，我们以PySpark运行SQL和sklearn求线性回归方程为例，介绍如何
```
Comment (hzjane) [inline docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md:56]: lack a blank line

File: docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+--cpuset-cpus 10-14 \
+--device=/dev/sgx/enclave \  #需提前配置好的sgx环境
+--device=/dev/sgx/provision \
+-v /var/run/aesmd:/var/run/aesmd \
+-v data:/opt/occlum_spark/data \
+-e SGX_MEM_SIZE=24GB \   #EPC即使用的SGX内存大小
+-e SGX_THREAD=1024 \
+-e SGX_HEAP=1GB \
+-e SGX_KERNEL_HEAP=1GB \
+-e SGX_LOG_LEVEL=off \intelanalytics/bigdl-ppml-trusted-big-data-ml-scala-occlum:2.2.0-SNAPSHOT \
```
Comment (hzjane) [inline docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md:86]: Divide codes and instructions

File: docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+bash
+2.	编写python源码，如sql_example.py 并将其放置在image的目录py-examples下
+3.	修改/opt/run_spark_on_occlum_glibc.sh文件，设置程序启动入口。
+run_pyspark_sql_example() {
+    init_instance spark  #执行occlum init初始化occlum文件结构并设置对应配置
+    build_spark #拷贝依赖并执行occlum build 构建可执行程序
+    cd /opt/occlum_spark
+    echo -e "${BLUE}occlum run pyspark SQL example${NC}"
+    occlum run /usr/lib/jvm/java-8-openjdk-amd64/bin/java \
+                -XX:-UseCompressedOops -XX:MaxMetaspaceSize=$META_SPACE \
```
Comment (hzjane) [inline docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md:96]: MaxMetaspaceSize is removed now，the two examples need to update.

File: docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md [hunk 4]
```diff
@@ -0,0 +1,509 @@
...
+    local:/py-examples/sklearn_example.py
+    # hdfs://ServerIP:Port/path/sklearn_example.py
+```
+
+注:若用云存储或者HDFS传入源文件则无需提前在image里传入源文件。
+
+## 3. 生产环境部署
+
+与快速上手阶段不同，生产部署需要考虑完整的数据流和密钥安全，并且需要根据现有的服务和设施进行对接。考虑到用户所用的服务有所差异，我们以开源和云服务为案例，介绍部署和配置KMS的基本过程；以及在安全环境中，构建生成环境中所需的image。
+安装和配置KMS
```
Comment (hzjane) [inline docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md:318]: header

File: docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md [hunk 5]
```diff
@@ -0,0 +1,509 @@
...
+  用PPMLContext和EHSM实现SimpleQuery应用的数据加解密流程
+1.	按照EHSM教程配置好PCCS和EHSM等环境。
+注意因为是部署在阿里云上，阿里云有可用的PCCS服务，所以对于教程里的第一步“Deploy BigDL-PCCS on Kubernetes”可以忽略。
+2.	注册获取app_id和api_key。
+```bash
+# Enroll
+curl -v -k -G "https://<kms_ip>:9000/ehsm?Action=Enroll"
+......
+
+{"code":200,"message":"successful","result":{"apikey":"E8QKpBBapaknprx44FaaTY20rptg54Sg","appid":"8d5dd3b8-3996-40f5-9785-dcb8265981ba"}}
```
Comment (hzjane) [inline docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md:334]: Divide codes and instructions

File: docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md [hunk 6]
```diff
@@ -0,0 +1,509 @@
...
+```
+
+5.	退出后提交得到最终的runnable image。 intelanalytics/bigdl-ppml-trusted-big-data-ml-scala-occlum-production: 2.2.0-build 即不添加任何外部依赖的runnable image，可直接运行任意已有的example。
+```bash
+docker commit $container_name $container_name-build
+```
+
+得到的未定制的intelanalytics/bigdl-ppml-trusted-big-data-ml-scala-occlum-production: 2.2.0-build大小有14.2GB，其中仅有/opt/occlum_spark文件夹和少部分配置文件是运行时所需的，其余大多数是拷贝和编译产生的垃圾文件。可在 production-build image的基础上copy occlum runnable instance 并安装Occlum运行时依赖和其他一些依赖得到最终的customer image，其大小仅不到5GB，且其功能与production-build image基本相同， intelanalytics/bigdl-ppml-trusted-big-data-ml-scala-occlum-production-customer: 2.2.0-build 即不经过任何定制的customer image。（通过修改运行build-customer-image.sh文件构建customer image）
+
+    Production-build 或 Customer image的attestation流程
```
Comment (hzjane) [inline docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md:455]: header

File: docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md [hunk 7]
```diff
@@ -0,0 +1,509 @@
...
+4.	在docker环境或者k8s环境设置policy_Id。
+```yaml
+#start-spark-local.sh
+-e ${policy_Id}
+
+#driver.yaml and executor.yaml
+env:
+  - name: policy_Id
+    value: "${policy_Id}"
+          5.	在docker或k8s启动应用（同上），仅会在SGX中运行EHSM对应用程序进行验证（IV. attest MREnclave）。
```
Comment (hzjane) [inline docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md:475]: Divide codes and instructions

File: docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md (hunk 3 above)
Comment (qiyuangong) [reply docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md:96]: Thank you for your comments! 
MaxMetaspaceSize removed from examples.

File: docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md (hunk 7 above)
Comment (qiyuangong) [reply docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md:475]: Addressed :)

File: docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md (hunk 2 above)
Comment (qiyuangong) [reply docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md:86]: Addressed

**Pull request merged**